RDKit test failure

RDKit test failures started to [show up in CI](https://github.com/MDAnalysis/mdanalysis/actions/runs/12105211274/job/33749526496?pr=4812
):
```
>               Chem.AssignStereochemistryFrom3D(mol)
E               RuntimeError: Cannot normalize a zero length vector
```


A new version of RDKit landed a couple of days ago: https://github.com/rdkit/rdkit/releases.

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -14,13 +14,14 @@ The rules for this file:
 
 
 -------------------------------------------------------------------------------
-??/??/?? IAlibay
+??/??/?? IAlibay, RMeli
 
  * 2.9.0
 
 Fixes
 
 Enhancements
+* Add check and warning for empty (all zero) coordinates in RDKit converter (PR #4824)
 
 Changes
 

--- a/package/MDAnalysis/converters/RDKit.py
+++ b/package/MDAnalysis/converters/RDKit.py
@@ -363,9 +363,12 @@ class RDKitConverter(base.ConverterBase):
 
         # add a conformer for the current Timestep
         if hasattr(ag, "positions"):
-            if np.isnan(ag.positions).any():
-                warnings.warn("NaN detected in coordinates, the output "
-                              "molecule will not have 3D coordinates assigned")
+            if np.isnan(ag.positions).any() or np.allclose(
+                ag.positions, 0.0, rtol=0.0, atol=1e-12
+            ):
+                warnings.warn("NaN or empty coordinates detected in coordinates, "
+                              "the output molecule will not have 3D coordinates "
+                              "assigned")
             else:
                 # assign coordinates
                 conf = Chem.Conformer(mol.GetNumAtoms())

--- a/testsuite/MDAnalysisTests/converters/test_rdkit.py
+++ b/testsuite/MDAnalysisTests/converters/test_rdkit.py
@@ -44,6 +44,7 @@ try:
                                              _standardize_patterns)
     from rdkit import Chem
     from rdkit.Chem import AllChem
+    from rdkit.Chem import rdDistGeom
 except ImportError:
     pass
 
@@ -682,6 +683,7 @@ class TestRDKitFunctions(object):
     def test_reorder_atoms(self, smi):
         mol = Chem.MolFromSmiles(smi)
         mol = Chem.AddHs(mol)
+        rdDistGeom.EmbedMolecule(mol)
         # remove bond order and charges info
         pdb = Chem.MolToPDBBlock(mol)
         u = mda.Universe(StringIO(pdb), format="PDB")

--- a/testsuite/MDAnalysisTests/converters/test_rdkit.py
+++ b/testsuite/MDAnalysisTests/converters/test_rdkit.py
@@ -44,7 +44,6 @@ try:
                                              _standardize_patterns)
     from rdkit import Chem
     from rdkit.Chem import AllChem
-    from rdkit.Chem import rdDistGeom
 except ImportError:
     pass
 
@@ -332,7 +331,7 @@ class TestRDKitConverter(object):
         xyz = u.atoms.positions
         xyz[0][2] = np.nan
         u.atoms.positions = xyz
-        with pytest.warns(UserWarning, match="NaN detected"):
+        with pytest.warns(UserWarning, match="NaN .* detected"):
             mol = u.atoms.convert_to("RDKIT")
         with pytest.raises(ValueError, match="Bad Conformer Id"):
             mol.GetConformer()
@@ -683,7 +682,6 @@ class TestRDKitFunctions(object):
     def test_reorder_atoms(self, smi):
         mol = Chem.MolFromSmiles(smi)
         mol = Chem.AddHs(mol)
-        rdDistGeom.EmbedMolecule(mol)
         # remove bond order and charges info
         pdb = Chem.MolToPDBBlock(mol)
         u = mda.Universe(StringIO(pdb), format="PDB")
@@ -693,6 +691,18 @@ class TestRDKitFunctions(object):
         values = [a.GetSymbol() for a in mu.GetAtoms()]
         expected = [a.GetSymbol() for a in mol.GetAtoms()]
         assert values == expected
+
+    @pytest.mark.parametrize("smi", [
+        "O=S(C)(C)=NC",
+    ])
+    def test_warn_empty_coords(self, smi):
+        mol = Chem.MolFromSmiles(smi)
+        mol = Chem.AddHs(mol)
+        # remove bond order and charges info
+        pdb = Chem.MolToPDBBlock(mol)
+        u = mda.Universe(StringIO(pdb), format="PDB")
+        with pytest.warns(match="NaN or empty coordinates detected"):
+            u.atoms.convert_to.rdkit()
 
     def test_pdb_names(self):
         u = mda.Universe(PDB_helix)

